internal/engine/buildcontroller.go

```diff
@@ -34,9 +34,14 @@ func NewBuildController(b BuildAndDeployer) *BuildController {
 
 // Algorithm to choose a manifest to build next.
 func nextTargetToBuild(state store.EngineState) *store.ManifestTarget {
+	// put no-build manifests first since they're more likely to be
+	// 1. fast and 2. dependencies of other services (e.g., redis)
+	targets := append([]*store.ManifestTarget{}, state.Targets()...)
```

Comment:
Does the Targets() method already make a new array so you don't need to clone here?

Reply:
Is that a detail that's worth relying on? I haven't seen much in the way of memory ownership semantics in go code, and worry mildly some later refactor might have `state.Targets()` returning the raw slice and now we're mutating `state`. I've often felt unsure whether it's appropriate to make a copy of a slice. Maybe I've been in languages with immutable collections for too long.

This change is already reducing the number of copies from 3 to 2, so I'm not too concerned about the performance implications for this change in particular, but I'm interested in the approach in general.